clients/src/main/java/org/apache/kafka/common/security/JaasUtils.java

```diff
@@ -58,6 +58,11 @@ public static boolean isZkSecurityEnabled(String loginConfigFile) {
                 throw new KafkaException("Exception while determining if ZooKeeper is secure");
             }
         }
+        /*
+         * Tests fail if we don't reset the login configuration. It is unclear
+         * what is actually triggering this bug.
+         */
+        Configuration.setConfiguration(null);
```

Comment:
yeah, I'm not comfortable with this at all, it doesn't sound robust. at the same time, I want to separate bugs in my code from bugs elsewhere, so I'm leaving this here for now, I need to understand what's causing this.